pykern/pkcollections.py

```diff
@@ -112,7 +173,15 @@ def pknested_get(self, dotted_key):
         """
         d = self
         for k in dotted_key.split("."):
-            d = d[k]
+            try:
```

Comment:
why not type check d instead of nesting try blocks?

Reply:
We don't know what `d` is, it's just something that responds to `__getitem__`. There is no coupling here except through TypeError which is the interface when `__getitem__` gets the wrong type of key. Then we try integer. If there's any particular error, we raise the original error so that it looks as it did before. This way we don't break abstractions.